float32[] words_histogram
float32[] shape_histogram
float32[] colors_histogram
float32[] hog_histogram
string[]  words_dictionary
